nifi-nar-bundles/nifi-extension-utils/nifi-processor-utils/src/test/java/org/apache/nifi/processor/util/list/TestAbstractListProcessor.java

```diff
@@ -105,6 +111,174 @@ public void setup() {
+        String timeAdjustment = "15";
+        long currentTime = 100;
+
+        long expected = 100 + 15;
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+
+    }
+
+    @Test
+    public void testGetAdjustedCurrentTimestampWithMinus15() throws Exception {
+        // GIVEN
+        String timeAdjustment = "-15";
+        long currentTime = 100;
+
+        long expected = 100 - 15;
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+
+    }
+
+    @Test
+    public void testGetAdjustedCurrentTimestampWithMinus_01_02() throws Exception {
+        // GIVEN
+        String timeAdjustment = "-01:02";
+        long currentTime = 100;
+
+        long expected = 100
+            - 1 * 60 * 60 * 1000
+            - 2 * 60 * 1000;
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+    }
+
+    @Test
+    public void testGetAdjustedCurrentTimestampWith_01_02() throws Exception {
+        // GIVEN
+        String timeAdjustment = "01:02";
+        long currentTime = 100;
+
+        long expected = 100
+            + 1 * 60 * 60 * 1000
+            + 2 * 60 * 1000;
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+    }
+
+    @Test
+    public void testGetAdjustedCurrentTimestampWith_01_02_34() throws Exception {
+        // GIVEN
+        String timeAdjustment = "01:02:34";
+        long currentTime = 100;
+
+        long expected = 100
+            + 1 * 60 * 60 * 1000
+            + 2 * 60 * 1000
+            + 34 * 1000;
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+    }
+
+    @Test
+    public void testGetAdjustedCurrentTimestampWithMinus_01_02_34() throws Exception {
+        // GIVEN
+        String timeAdjustment = "-01:02:34";
+        long currentTime = 100;
+
+        long expected = 100
+            - 1 * 60 * 60 * 1000
+            - 2 * 60 * 1000
+            - 34 * 1000;
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+    }
+
+    @Test
+    public void testGetAdjustedCurrentTimestampWithEST() throws Exception {
+        // GIVEN
+        String timeAdjustment = "EST";
+        long currentTime = System.currentTimeMillis();
+
+        TimeZone targetTimeZone = TimeZone.getTimeZone(timeAdjustment);
+        TimeZone localTimeZone = Calendar.getInstance().getTimeZone();
+        long expected = currentTime + targetTimeZone.getOffset(currentTime) - localTimeZone.getOffset(currentTime);
+
+        // WHEN
+        // THEN
+        testGetAdjustedCurrentTimestamp(timeAdjustment, expected, currentTime);
+    }
+
+    private void testGetAdjustedCurrentTimestamp(String timeAdjustment, long expected, final long currentTime) {
```

Comment:
As there are a lot of abstract method needs to be implemented, it might be a possibility to use Mockito.spy